### PR TITLE
add support for jaeger tracing

### PR DESCRIPTION
**Pull request by adilhafeez**
- move prompt and llm gateway wasm filter into single listener
- create trace_arch_internal param so we can expose arch internal spans for local development
- when making outbound calls from filter attach traceparent in the header
- create a new internal listener to generate traceparent so spans from filter can be grouped properly
- create signoz and jaeger shared service
- rename function_calling to weather_forecast

trace ui for a sample request,

image

### CODE REVIEW

File: arch/arch_config_schema.yaml [hunk 1]
```diff
@@ -149,6 +149,8 @@ properties:
     properties:
       random_sampling:
         type: integer
+      trace_arch:
```
Comment (salmanap) [inline arch/arch_config_schema.yaml:152]: trace_arch_internal please

File: demos/shared/insurance_claims_service/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# took inspiration from https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0
```
Comment (salmanap) [inline demos/shared/insurance_claims_service/Dockerfile:1]: Please remove this altogether. No need to keep this insurance_claims_service. We have a separate demo for that

File: demos/shared/insurance_claims_service/main.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import json
```
Comment (salmanap) [inline demos/shared/insurance_claims_service/main.py:1]: Please remove this altogether. See comment above

File: demos/shared/insurance_claims_service/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[tool.poetry]
```
Comment (salmanap) [inline demos/shared/insurance_claims_service/pyproject.toml:1]: Remove please. See above

**Review by salmanap — COMMENTED** [review]
I

File: demos/shared/insurance_claims_service/main.py (hunk 3 above)
Comment (adilhafeez) [reply demos/shared/insurance_claims_service/main.py:1]: removed

File: demos/shared/insurance_claims_service/Dockerfile (hunk 2 above)
Comment (adilhafeez) [reply demos/shared/insurance_claims_service/Dockerfile:1]: removed

File: demos/shared/insurance_claims_service/pyproject.toml (hunk 4 above)
Comment (adilhafeez) [reply demos/shared/insurance_claims_service/pyproject.toml:1]: removed

File: arch/arch_config_schema.yaml (hunk 1 above)
Comment (adilhafeez) [reply arch/arch_config_schema.yaml]: updated

File: model_server/app/main.py [hunk 5]
```diff
@@ -15,12 +16,42 @@
 )
 from unittest.mock import patch
 
+from opentelemetry import trace
```
Comment (salmanap) [inline model_server/app/main.py:19]: Please remove tracing from model_server internal calls. Not necessary

Comment (adilhafeez) [reply model_server/app/main.py:19]: its not used - see this line https://github.com/katanemo/arch/pull/229/files#diff-a883497ecfcfe46bc8a13b33b682d89dc175ff777398e7f8fce68cfc990087f1R46

**Review by salmanap — APPROVED** [review]

**Pull request merged**